libraries/cbor/src/values.rs

```diff
@@ -315,12 +319,17 @@ mod test {
         assert!(cbor_int!(-1) < cbor_text!("s"));
         assert!(cbor_int!(-1) < cbor_array![]);
         assert!(cbor_int!(-1) < cbor_map! {});
+        assert!(cbor_int!(-1) < cbor_tagged!(1, cbor_text!("s")));
         assert!(cbor_int!(-1) < cbor_bool!(false));
         assert!(cbor_bytes!(vec![0x00]) < cbor_array![]);
         assert!(cbor_bytes!(vec![0x00]) < cbor_map! {});
         assert!(cbor_bytes!(vec![0x00]) < cbor_bool!(false));
+        assert!(cbor_bytes!(vec![0x00]) < cbor_tagged!(1, cbor_text!("s")));
         assert!(cbor_text!("s") < cbor_map! {});
         assert!(cbor_text!("s") < cbor_bool!(false));
+        assert!(cbor_text!("s") < cbor_tagged!(1, cbor_text!("s")));
         assert!(cbor_array![] < cbor_bool!(false));
+        assert!(cbor_tagged!(1, cbor_text!("s")) < cbor_tagged!(2, cbor_int!(0)));
+        assert!(cbor_tagged!(1, cbor_text!("s")) < cbor_bool!(false));
```

Comment:
Can you add a major type check between map and tag?

Reply:
Done (plus a couple of other comparisons and a slight re-order to match major type order a bit better)